app/src/main/java/com/xmartlabs/gong/ui/theme/AppTypographies.kt

```diff
@@ -0,0 +1,110 @@
+@file:Suppress("unused", "MagicNumber")
+
+package com.xmartlabs.gong.ui.theme
+
+import androidx.compose.material.Typography
+import androidx.compose.runtime.Immutable
+import androidx.compose.runtime.Stable
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.Font
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.text.font.FontStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.sp
+import com.xmartlabs.gong.R
+
+/**
+ * Created by mirland on 26/3/21.
+ */
+@Immutable
+data class AppTypography(
+    val primaryButton: TextStyle,
+    val linkButton: TextStyle,
+
+    val materialTypography: Typography
+) {
+  val h1: TextStyle
+    get() = materialTypography.h1
+
+  val h2: TextStyle
+    get() = materialTypography.h2
+
+  val h3: TextStyle
+    get() = materialTypography.h3
+
+  val h4: TextStyle
+    get() = materialTypography.h4
+
+  val h5: TextStyle
+    get() = materialTypography.h5
+
+  val h6: TextStyle
+    get() = materialTypography.h6
+
+  val subtitle1: TextStyle
+    get() = materialTypography.subtitle1
+
+  val subtitle2: TextStyle
+    get() = materialTypography.subtitle2
+
+  val body1: TextStyle
+    get() = materialTypography.body1
+
+  val body2: TextStyle
+    get() = materialTypography.body2
+
+  val button: TextStyle
+    get() = materialTypography.button
+
+  val caption: TextStyle
+    get() = materialTypography.caption
+
+  val overline: TextStyle
+    get() = materialTypography.overline
+}
+
+fun defaultAppTypography() = appTypography(defaultAppDims(), defaultAppColors())
+
+fun appTypography(dims: AppDims, colors: AppColors): AppTypography {
+  val baseTextStyle = TextStyle(
+      fontFamily = AppFontFamilies.Roboto,
+      fontWeight = FontWeight.Normal,
+      fontSize = dims.textSizeRegular,
+      letterSpacing = (-1.5).sp,
+  )
+
```

Comment:
```suggestion

fun defaultAppTypography() = appTypography()

fun appTypography(dims: AppDims = defaultAppDims(), colors: AppColors = defaultAppColors()): AppTypography {
  val baseTextStyle = TextStyle(
      fontFamily = AppFontFamilies.Roboto,
      fontWeight = FontWeight.Normal,
      fontSize = dims.textSizeRegular,
      letterSpacing = (-1.5).sp,
  )

```

Maybe it's easier to create more typographies this way, maybe we want to change just the colors or just the dims